Dialog/RoomConfigDialog.cpp

```diff
@@ -0,0 +1,227 @@
+#include "RoomConfigDialog.h"
+#include "ui_RoomConfigDialog.h"
+
+RoomConfigDialog::RoomConfigDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::RoomConfigDialog)
+{
+    ui->setupUi(this);
+}
+
+RoomConfigDialog::~RoomConfigDialog()
+{
+    delete ui;
+}
+
+RoomConfigDialog::InitDialog()
+{
+    InitComboBoxItems();
+    ui->ComboBox_AlphaBlendAttribute->setCurrentIndex(0);
+    ui->ComboBox_Layer0MappingType->setCurrentIndex(0);
+    // TODO
+
+}
+
+RoomConfigDialog::InitDialog(DialogParams::RoomConfigParams CurrentRoomParams)
+{
+    InitComboBoxItems();
+    ui->ComboBox_TilesetID->setCurrentIndex(CurrentRoomParams.CurrentTilesetIndex);
+    ui->CheckBox_Layer0Enable->setChecked(CurrentRoomParams.Layer0Enable);
+    ui->CheckBox_Layer0Alpha->setChecked(CurrentRoomParams.Layer0Alpha);
+    int LayerPriorityID = CurrentRoomParams.LayerPriorityAndAlphaAttr & 3;
+    ui->ComboBox_LayerPriority->setCurrentIndex((LayerPriorityID < 2) ? LayerPriorityID : (LayerPriorityID - 1));
+    int AlphaBlendID = (CurrentRoomParams.LayerPriorityAndAlphaAttr & 0x78) >> 3;
+    ui->ComboBox_AlphaBlendAttribute->setCurrentIndex(AlphaBlendID);
+    if(CurrentRoomParams.Layer0Enable) ui->ComboBox_Layer0MappingType->setCurrentIndex(((CurrentRoomParams.Layer0MappingTypeParam) & 0x30) >> 4);
+    if(CurrentRoomParams.Layer0MappingTypeParam == 0x22) ui->CheckBox_Layer0Scrolling->setChecked(true);
+    ui->SpinBox_RoomWidth->setValue(CurrentRoomParams.RoomWidth);
+    ui->SpinBox_RoomHeight->setValue(CurrentRoomParams.RoomHeight);
+    ui->CheckBox_Layer2Enable->setChecked(CurrentRoomParams.Layer2Enable);
+    ui->CheckBox_BGLayerEnable->setChecked(CurrentRoomParams.BackgroundLayerEnable);
+    ui->CheckBox_BGLayerScrolling->setChecked(CurrentRoomParams.BackgroundLayerScrollingEnable);
+    // TODO
```

Comment:
We need to set the currentParams struct. If you are using a pointer for the input param, you can use memcpy(&currentParams, CurrentRoomParams, sizeof(struct DialogParams::RoomConfigParams))